<!-- RECURSIVE LAUNCH FILE TO INITIALIZE ALL THE NODES IN THE CORRECT NAMESPACE -->

<!-- TODO: check if the recursion is correct -->
<launch>

    <!-- ARGUMENT VALUE HOLDING THE TOTAL NUMBER OF ROBOTS, IT WILL BE PASSED FROM THE PYTHON SCRIPT -->
    <arg name="num_drones" default="1" />
    <arg name="nd" default="$(arg num_drones)" />
    <!-- DRONE NODES -->
        <group ns="$(eval 'drone' + str(arg('nd')))">
        <arg name="drone_model" default="/home/marco/shared/working_dir/catkin_ws/src/hector_quadrotor_noetic/hector_quadrotor/hector_quadrotor_description/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro" />
<!--         <include file="$(find dist_project)/launch/yaya.launch"></include> -->
<!--         <param name="drone_description" command="$(find xacro)/xacro $(arg drone_model)" /> -->

<!--           <node name="spawn_robot" pkg="gazebo_ros" type="spawn_model" -->
<!--             args="-param drone_description -->
<!--                -urdf -->
<!--                -x 0 -->
<!--                -y 0 -->
<!--                -z 0 -->
<!--                -model droninho" -->
<!--             respawn="false" output="screen"/> -->

<!--            <param name="base_link_frame" default="base_link"/> -->
<!--             <arg name="robot_description" default="/home/marco/shared/working_dir/catkin_ws/src/hector_quadrotor_noetic/hector_quadrotor/hector_quadrotor_description/urdf/strana_bestia.urdf" /> -->
<!--             <param name="robot_description" command="$(find xacro)/xacro '$(arg drone_model)' base_link_frame:=base_link world_frame:=world" /> -->

<!--             <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen" > -->
<!--                 <param name="publish_frequency" type="double" value="50.0"  /> -->
<!--                 <param name="robot_description" value="$(arg robot_description)" /> -->
<!--           </node> -->

<!--           <node name="ground_truth_to_tf" pkg="message_to_tf" type="message_to_tf" output="screen"> -->
<!--             <param name="odometry_topic" value="ground_truth/state" /> -->
<!--             <param name="frame_id" value="world" /> -->
<!--             <param name="tf_prefix" value="" /> -->
<!--           </node> -->


<!--           <group> -->
<!--             <param name="controller/state_topic" value="" /> -->
<!--             <param name="controller/imu_topic" value="" /> -->
<!--           </group> -->
<!--           <include file= "$(find hector_quadrotor_controller)/launch/controller.launch" /> -->

<!--           <arg name="motors" default="robbe_2827-34_epp1045" /> -->
<!--           <rosparam command="load" file="$(find hector_quadrotor_model)/param/quadrotor_aerodynamics.yaml" /> -->
<!--           <rosparam command="load" file="$(find hector_quadrotor_model)/param/$(arg motors).yaml" /> -->
<!--             <node name="uwb_dist_sim_drone" pkg="dist_project" type="uwb_dist_sim_drone.py" /> -->
<!--             <node name="altitude_stabilizer_drone" pkg="dist_project" type="altitude_stabilizer_drone.py" /> -->
<!--             <node name="motion_planner_drone" pkg="dist_project" type="motion_planner_drone.py" /> -->
<!--             <node name="kalman_localization_drone" pkg="dist_project" type="kalman_localization_drone.py" /> -->
<!--             <node name="image_processing_drone" pkg="dist_project" type="image_processing_drone.py" /> -->
<!--             <node name="target_estimator_drone" pkg="dist_project" type="target_estimator_drone.py"> -->
<!--                 <param name="namespace" value="$(arg nd)" /> -->
<!--             </node> -->
        </group>
        <include file="$(find dist_project)/launch/start_drone_nodes.launch" if="$(eval arg('nd') - 1 >= 0)">
            <arg name="num_drones" value="$(eval arg('nd') - 1)" />
        </include>

</launch>